pandas/plotting/_core.py

```diff
@@ -1398,6 +1398,10 @@ def _get_ind(self, y):
             sample_range = np.nanmax(y) - np.nanmin(y)
             ind = np.linspace(np.nanmin(y) - 0.5 * sample_range,
                               np.nanmax(y) + 0.5 * sample_range, 1000)
+        elif isinstance(self.ind, (int, np.int)):
```

Comment:
Use  `isinstance(self.ind, numbers.Integral)`. It should catch both of these (may have to import `numbers`).

Reply:
instead of this, just use ``is_integer``, which is our version of this and handles all the cases.
import from ``pandas.core.dtypes.common`` (if not already imported)